src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -2281,5 +2281,723 @@ public class D { }
+            >50 => ""more"",
+        }});
+    }}
+}}
+";
+                var expectedOutput =
+    @"less
+less
+same
+more";
+                var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview), options: TestOptions.DebugExe);
+                compilation.VerifyDiagnostics(
+                    );
+                var verifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            }
+        }
+
+        [Fact]
+        public void Relational_04()
+        {
+            var source = @"
+class C
+{
+    static void M1(object o, decimal d)
+    {
+        _ = o is < 12m;
+        _ = d is < 0;
+        _ = o is < 10;
+        switch (d)
+        {
+            case < 0m:
+            case <= 0m:
+            case > 0m:
+            case >= 0m:
+                break;
+        }
+    }
+}
+";
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+            compilation.VerifyDiagnostics(
+                // (6,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //         _ = o is < 12m;
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, "< 12m").WithArguments("decimal").WithLocation(6, 18),
+                // (7,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //         _ = d is < 0;
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, "< 0").WithArguments("decimal").WithLocation(7, 18),
+                // (11,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //             case < 0m:
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, "< 0m").WithArguments("decimal").WithLocation(11, 18),
+                // (12,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //             case <= 0m:
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, "<= 0m").WithArguments("decimal").WithLocation(12, 18),
+                // (13,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //             case > 0m:
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, "> 0m").WithArguments("decimal").WithLocation(13, 18),
+                // (14,18): error CS8781: Relational patterns may not be used for a value of type 'decimal'.
+                //             case >= 0m:
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForRelationalPattern, ">= 0m").WithArguments("decimal").WithLocation(14, 18)
+                );
+        }
+
+        [Fact]
+        public void Relational_05()
+        {
+            foreach (string typeName in new[] { "byte", "sbyte", "short", "ushort", "int" })
```

Comment:
>} [](start = 90, length = 1)

`, "char"

Reply:
This is a list of types that are compared by promotion to the type `int`.

---
In reply to: [383046028](https://github.com/dotnet/roslyn/pull/41503#discussion_r383046028) [](ancestors = 383046028)